Subject: [PATCH 01/19] ieee1394: sbp2: enable auto spin-up for Maxtor disks
From: Stefan Richter <stefanr@s5r6.in-berlin.de>
References: 183011

At least Maxtor OneTouch III require a "start stop unit" command after
auto spin-down before the next access can proceed.  This patch activates
the responsible code in scsi_mod for all Maxtor SBP-2 disks.
https://bugzilla.novell.com/show_bug.cgi?id=183011

Maybe that should be done for all SBP-2 disks, but better be cautious.

Signed-off-by: Stefan Richter <stefanr@s5r6.in-berlin.de>
Signed-off-by: Olaf Hering <olh@suse.de>

---
 drivers/ieee1394/sbp2.c |    3 +++
 1 file changed, 3 insertions(+)

Index: linux-2.6.16/drivers/ieee1394/sbp2.c
===================================================================
--- linux-2.6.16.orig/drivers/ieee1394/sbp2.c
+++ linux-2.6.16/drivers/ieee1394/sbp2.c
@@ -2514,6 +2514,9 @@ static int sbp2scsi_slave_configure(stru
 		SBP2_INFO("enabling iPod workaround: decrement disk capacity");
 		sdev->fix_capacity = 1;
 	}
+	if (scsi_id->ne->guid_vendor_id == 0x0010b9 && /* Maxtor's OUI */
+		(sdev->type == TYPE_DISK || sdev->type == TYPE_RBC))
+			sdev->allow_restart = 1;
 	return 0;
 }
 
